tests/memory_checker/test_memory_checker.py

```diff
@@ -420,238 +373,289 @@ def get_container_mem_usage(duthost, container_name):
+
+    def get_last_start_date(self):
+        rc = self.duthost.shell(r"docker inspect --format \{{\{{.State.StartedAt\}}\}} {}".format(self.name))
+        date_str = rc['stdout_lines'][0].strip()
+        return dateutil.parser.isoparse(date_str)
+
+    def has_container_restarted(self):
+        start_date = self.get_last_start_date()
+        return start_date > self._last_start_date
+
+    def wait_restarted(self, start_date=None):
+        self._last_start_date = start_date or self.get_last_start_date()
+        restarted = wait_until(CONTAINER_RESTART_THRESHOLD_SECS, 1, 0,
+                               self.has_container_restarted)
+        pytest_assert(restarted, "Failed to restart '{}' container!".format(self.name))
+        self._last_start_date = None
+
+    def wait_monit_mem_ok(self, timeout=MONIT_MEMORY_CHECK_TIMEOUT):
+        logger.info("Waiting for monit status ok for %s", self.name)
+        res = wait_until(timeout, 1, 0, self.is_monit_mem_ok)
+        pytest_assert(res, "Failed to wait for one monit cycle to be ok for {}".format(self.name))
+
+    def wait_monit_mem_failed(self, timeout=MONIT_MEMORY_CHECK_TIMEOUT):
+        logger.info("Waiting for monit status failed for %s", self.name)
+        res = wait_until(timeout, 1, 0, self.is_monit_mem_failed)
+        pytest_assert(res, "Failed to wait for one monit cycle to fail for {}".format(self.name))
+
+    def wait_monit_mem_last_ok(self, timeout=MONIT_MEMORY_CHECK_TIMEOUT):
+        logger.info("Waiting for last monit status ok for %s", self.name)
+        res = wait_until(timeout, 1, 0, self.is_monit_mem_last_ok)
+        pytest_assert(res, "Failed to wait for one monit cycle to be ok for {}".format(self.name))
+
+    def wait_monit_mem_last_failed(self, timeout=MONIT_MEMORY_CHECK_TIMEOUT):
+        logger.info("Waiting for last monit status failed for %s", self.name)
+        res = wait_until(timeout, 1, 0, self.is_monit_mem_last_failed)
+        pytest_assert(res, "Failed to wait for one monit cycle to fail for {}".format(self.name))
+
+    def wait_ready(self):
+        if not self.is_running():
+            pytest.fail("'{}' is not running!".format(self.name))
+        self.post_check()
+        self.wait_monit_mem_ok()
+
+
+def consumes_memory_and_checks_container_restart(duthost, container):
+    """Allocates memory in the container and checks whether the container can be
+    stopped and restarted. Loganalyzer is leveraged to check whether the log messages
+    related to container stopped were generated.
+
+    Args:
+        duthost: The AnsibleHost object of DuT.
+        container: Container object to test
+
+    Returns:
+        None.
+    """
+    marker_prefix = "container_restart_due_to_memory"
+    loganalyzer = LogAnalyzer(ansible_host=duthost, marker_prefix=marker_prefix)
+    loganalyzer.expect_regex = container.get_restart_expected_logre()
+    with loganalyzer:
+        timeout_monit_fail = 60  # fails happens after 10 cycles of 1 second
```

Comment:
would 60sec be enough for chassis as well?

Reply:
Yes, this test is completely topology agnostic.
It only tests the memory checker for the telemetry container.
This test should therefore behave exactly the same on any switch.

As the inline comment mentions, the test runs with the monit daemon configured with no startup delay and a cycle time of 1 second.
The `monit_telemetry` configuration is unchanged and would therefore fail after 10 cycles of 1 second.
60 is 6x the time necessary for monit to correctly poll and update the status so that should be plenty.